rplugin/python3/denite/source/menu.py

```diff
@@ -0,0 +1,68 @@
+# ============================================================================
+# FILE: menu.py
+# AUTHOR: Shougo Matsushita <Shougo.Matsu at gmail.com>
+#         TJ DeVries <devries.timothyj at gmail.com>
+# License: MIT license
+# ============================================================================
+
+from .base import Base
+
+
+class Source(Base):
+
+    def __init__(self, vim):
+        Base.__init__(self, vim)
+
+        self.name = 'menu'
+        self.kind = 'jump_list'
+
+        # self.matchers = []
+        # self.sorters = []
+
+        self.dict_key = '__menus'
+
+    def on_init(self, context):
+        context[self.dict_key] = context['custom']['source']['menu']['vars']\
```

Comment:
You don't have to set the context.
self.vars is overwritten by `denite#custom#var()`.